src/PyDictOrObjectProxyHandler.cc

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file PyDictOrObjectProxyHandler.cc
```

Comment:
This class name is a bit awkward. Does it make sense to have Dict be a subclass of Object instead? If not, can we look into the Python language spec for name inspiration?

Reply:
till seperate PR